vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -4663,6 +4665,31 @@ public void scrollToIndex(int rowIndex) {
         getElement().callJsFunction("scrollToIndex", rowIndex);
     }
 
+    /**
+     * Scrolls to the row presenting the given item.
+     * <p>
+     * Note that the item index provider should be explicitly set using
+     * {@link GridLazyDataView#setItemIndexProvider(ItemIndexProvider)} for lazy
+     * loading data providers.
+     *
+     * @param item
+     *            the item to scroll to, not {@code null}.
+     * @throws NullPointerException
+     *             if the {@code item} parameter is {@code null}.
+     * @throws NoSuchElementException
+     *             if the {@code item} cannot be found.
+     */
+    public void scrollToItem(T item) {
+        Objects.requireNonNull(item, "Item to scroll to cannot be null.");
+        AbstractDataView<T> dataView = getDataProvider().isInMemory()
+                ? getListDataView()
+                : getLazyDataView();
```

Comment:
The current tests pass even if we use `int itemIndex = getGenericDataView().getItemIndex(item)` here. Please add a test case that uses `setItemIndexProvider` to validate this check.

Reply:
Added a test for a missing item index provider. It fails when `getGenericDataView()` is used.